migrations/v7-MIGRATION.md

```diff
@@ -0,0 +1,43 @@
+# V7 API Migration Guide
+
+This latest release updates the SDK to use BillingClient 6. This version of BillingClient brings an entire new
+subscription model which has resulted in large changes across the entire SDK.
```

Comment:
> This version of BillingClient brings an entire new
subscription model which has resulted in large changes across the entire SDK.

Copy/pasta from BC5?

Reply:
Also it'd be great to provide a high level here about the news in BC6, which aren't plenty - it's mostly updates to product change purchases and deferred purchases, right?

Reply:
I've added a better explanation of the BC6 changes plus the deactivation of the DEFERRED mode to the migration guide.